raijin_scripts/execute_s2ard/run

```diff
@@ -60,16 +60,21 @@ if [ -z "$FILE_MOD_END" ]; then
 fi
 
 if [ "$OBS_YEAR" == "current" ]; then
-    # uses seq to handle end of year periods where it would be ideal to process both years
-    OBS_YEAR="$( seq "$(date -d"$FILE_MOD_START" +"%Y")" "$(date -d"$FILE_MOD_END" +"%Y")")"
+    # Calculates start and end years from the observation window to hand Dec - Jan window
+    _OBS_YEAR_START="$(date -d"$FILE_MOD_START" +"%Y")"
+    _OBS_YEAR_END="$(date -d"$FILE_MOD_END" +"%Y")"
 elif [ "$OBS_YEAR" == "previous" ]; then
-    OBS_YEAR="$( seq $S2_START_YEAR "$(date -d"$FILE_MOD_END - 1 year" +"%Y")" )"
-elif [ -n "$OBS_YEAR" ]; then
-    # Turn the year into an array
-    OBS_YEAR="$( seq "$OBS_YEAR" "$OBS_YEAR" )"
+    # Process all prior years; there should be less new datasets from previous periods
+    _OBS_YEAR_START="${S2_START_YEAR}"
+    _OBS_YEAR_END="$(date -d"$FILE_MOD_END - 1 year" +"%Y")"
+elif [ -z "${OBS_YEAR}" ]; then
+    # If defined just process that year
+    _OBS_YEAR_START="${OBS_YEAR}"
+    _OBS_YEAR_END="${OBS_YEAR}"
 else
```

Comment:
The condition is evaluating for no string length for `"${OBS_YEAR}"`.

Is this else if condition processing year specified in the serverless config file? If so, are we after the following `elif` condition?
`elif [ date -d "${OBS_YEAR}" >/dev/null 2>&1 ]`

Reply:
thanks for picking this up